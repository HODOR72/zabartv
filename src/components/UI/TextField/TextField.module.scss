.textField {
	-webkit-appearance: none;
	appearance: none;
	outline: none;
	background-color: #fff;
	border-radius: 5px;
	padding-left: 30px;
	padding-right: 30px;
	border: none;
	height: 57px;
	font-weight: 700;
	font-size: 16px;
	padding-top: 4px;
	line-height: 28.8px;
	border-radius: 50px;
	border: 1px solid transparent;
	color: #0b1527;
	transition: background-color 0.3s, color 0.3s, border-color 0.3s;

	&:not(.error) {
		&:focus,
		&:active {
			border-color: #ccd5e8;
		}
	}

	&.error {
		border-color: var(--color-red);
		color: var(--color-red);
		margin-bottom: 0;

		&:not(:last-child) {
			margin-bottom: 0;
		}

		&::placeholder {
			color: var(--color-red);
		}
	}

	&::-webkit-search-decoration,
	&::-webkit-search-cancel-button,
	&::-webkit-search-results-button,
	&::-webkit-search-results-decoration {
		display: none;
	}

	&::placeholder {
		font-weight: 500;
		font-size: 20px;
		line-height: 36px;
		color: #d2d6e2;
		opacity: 1;
	}
}

[data-theme='light'] {
	.dark {
		background-color: #f5f6f7;
		color: #0b1527;
	}
}

.dark {
	background-color: #232c3c;
	font-weight: 700;
	font-size: 16px;
	line-height: 28.8px;
	color: #fff;
	
	&:not(.error) {
		&:focus,
		&:active {
			border-color: transparent;
		}
	}

	&::placeholder {
		color: #a5a5ad;
	}
}

.errorMessage {
	display: block;
	font-size: 16px;
	line-height: 28.8px;
	font-weight: 500;
	color: var(--color-red);
	margin-top: 6px;

	@media (max-width: 576px) {
		font-size: 14px;
		line-height: 25px;
	}

	&:not(:last-child) {
		margin-bottom: 7px;
	}
}
