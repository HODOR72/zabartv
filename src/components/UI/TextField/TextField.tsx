import { forwardRef, InputHTMLAttributes } from 'react';
import classNames from 'classnames';
import styles from './TextField.module.scss';

export interface TextFieldProps extends InputHTMLAttributes<HTMLInputElement> {
	variant?: 'dark';
	errorMessage?: string;
	error?: boolean;
}

export const TextField = forwardRef<HTMLInputElement, TextFieldProps>(
	({ className, errorMessage, error, variant, ...props }, ref) => {
		return (
			<>
				<input
					data-testid="input"
					ref={ref}
					className={classNames(
						styles.textField,
						variant === 'dark' && styles.dark,
						error && styles.error,
						className
					)}
					{...props}
				/>
				{errorMessage && <span className={styles.errorMessage}>{errorMessage}</span>}
			</>
		);
	}
);

TextField.displayName = 'TextField';
