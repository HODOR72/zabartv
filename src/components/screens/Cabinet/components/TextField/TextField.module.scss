.field {
    position: relative;
    border: 1px solid #232C3D;
    border-radius: 25px;
    padding: 32px 45px 37px;
    transition: border-color 0.3s;

    @media (max-width: 1200px) {
        padding: 27px 30px;
    }

    @media (max-width: 768px) {
        padding: 32px 45px 37px;
    }
}

[data-theme="light"] {
    .field {
        border-color: #e2E0dd;
    }
}

.label {
    display: block;
    font-weight: 500;
    font-size: 16px;
    line-height: 27.8px;
    color: #52596C;
}

.input {
    -webkit-appearance: none;
    width: 100%;
    appearance: none;
    padding: 0;
    border: none;
    border-radius: 0;
    background-color: transparent;
    outline: none;
    font-family: inherit;
    color: var(--color-light);
    margin-top: 5px;
    transition: color 0.3s;

    &::placeholder {
        color: #52596C;
    }
}

.value {
    pointer-events: none;
    cursor: default;
    margin-top: 0;

    &[type='password'] {
        font-family: Verdana;
        letter-spacing: 0.5px;
    }
}

.apply {
    margin-top: 5px;
}

.btn {
    position: absolute;
    display: block;
    right: 23px;
    top: 23px;
    width: 18px;
    height: 18px;

    svg {
        display: block;
        width: 100%;
        height: 100%;
        fill: var(--color-white);
    }
}