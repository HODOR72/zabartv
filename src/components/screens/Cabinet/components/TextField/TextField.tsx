import { ChangeEvent, FC, FormEvent, InputHTMLAttributes, useRef, useState } from 'react'
import { ButtonBase } from '@/components/ButtonBase/ButtonBase';
import { EditIcon, CloseIcon } from '@/icons'
import { Link } from '@/UI/Link/Link';
import classNames from 'classnames'
import styles from './TextField.module.scss'

export interface TextFieldProps extends InputHTMLAttributes<HTMLInputElement> {
    label: string;
    date?: boolean;
    mask?: string;
    maskChar?: string;
    value: string;
    as?: 'input' | 'mask';
    onChange:any;
}

export const TextField: FC<TextFieldProps> = ({className, onChange, type, value, as = 'input', date, label, ...props}) => {

    const inputRef = useRef<HTMLInputElement>(null)
    const [isEdited, setIsEdited] = useState<boolean>(false)
    const [inputValue, setInputValue] = useState(value)

    const closeField = () => {
        inputRef.current?.blur()
        setIsEdited(false)
    }

    const handleEditField = () => {
        if(!isEdited) {
            setIsEdited(true)
            inputRef.current?.focus()
        } else {
            setInputValue(value)
            closeField()
        }
    }

    const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
        setInputValue(e.target.value)
    }

    const applyChanges = () => {
        onChange(inputValue)
        closeField()
    }

    return (
        <div className={styles.field}>
            <span className={styles.label}>{label}</span>
            <input
                ref={inputRef}
                className={classNames(styles.input, !isEdited && styles.value)}
                type={isEdited ? 'text' : type}
                placeholder={value}
                value={inputValue}
                onChange={handleChange}
                {...props}
            />
            <ButtonBase type='button' onClick={handleEditField} className={styles.btn}>
                {!isEdited ? <EditIcon /> : <CloseIcon />}
            </ButtonBase>
            {isEdited && <Link as='button' onClick={applyChanges} className={styles.apply}>Сохранить изменения</Link>}
        </div>
    )
}