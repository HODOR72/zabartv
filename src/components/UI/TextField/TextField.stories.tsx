import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { TextField } from './TextField';

export default {
	title: 'TextField',
	component: TextField,
} as ComponentMeta<typeof TextField>;

const Template: ComponentStory<typeof TextField> = (args) => <TextField {...args} />;

export const White = Template.bind({});
White.args = {
	placeholder: 'Some placeholder',
};

export const Dark = Template.bind({});
Dark.args = {
	placeholder: 'Some placeholder',
	variant: 'dark',
};
