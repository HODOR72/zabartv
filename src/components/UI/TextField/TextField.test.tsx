import { TextField } from './TextField';
import { fireEvent, render, screen } from '@testing-library/react';

describe('TextField.test.tsx', () => {
	test('tests', () => {
		render(<TextField />);
		const element = screen.getByTestId('input');
		expect(element).toBeInTheDocument();
		expect(element).toHaveValue('');
		fireEvent.input(element, {
			target: { value: 'Some text' },
		});
		expect(element).toHaveValue('Some text');
		expect(element.tagName).toEqual('INPUT');
		expect(element).toMatchSnapshot();
	});
});
