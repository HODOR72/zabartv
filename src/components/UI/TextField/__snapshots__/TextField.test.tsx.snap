// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField.test.tsx tests 1`] = `
<input
  class="textField"
  data-testid="input"
/>
`;
